crates/evm/src/query.rs

```diff
@@ -348,13 +349,23 @@ impl<C: sov_modules_api::Context> Evm<C> {
         block_id: Option<BlockId>,
         working_set: &mut WorkingSet<C>,
     ) -> RpcResult<reth_primitives::Bytes> {
+        let cfg = self
+            .cfg
+            .get(working_set)
+            .expect("EVM chain config should be set");
+        let (_, current_spec) = cfg.spec.last().expect("Spec should be set");
+
         self.set_state_to_end_of_evm_block_by_block_id(block_id, working_set)?;
 
         let account = self.accounts.get(&address, working_set).unwrap_or_default();
         let code = if let Some(code_hash) = account.code_hash {
-            self.code
-                .get(&code_hash, &mut working_set.offchain_state())
-                .unwrap_or_default()
+            if current_spec.is_enabled_in(CANCUN) {
+                self.offchain_code
+                    .get(&code_hash, &mut working_set.offchain_state())
+                    .unwrap_or_default()
```

Comment:
code might have not been migratted, so if cancun is enabled but the code cant be found in offchain code, try to get it from old code (you cant migrate it here tho, so just get)